### ISSUE TITLE
Docker is crashing once switched to Windows containers

### ISSUE THREAD

**Issue by dreamflame51** [docker/for-win#1585]
System: Windows Server 2016 Datacenter 10.0.14393 1607 on bare metal
Docker: CE 17.12.0-ce-win47 (15139) stable

Once switching to Windows Containers following error occurred:
Error response from daemon: open \.\pipe\docker_engine_windows: The system cannot find the file specified.

at Docker.Backend.DockerDaemonChecker.Check(Func`1 isDaemonProcessStillRunning) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\DockerDaemonChecker.cs:line 63
at Docker.Backend.ContainerEngine.Windows.DoStart(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Windows.cs:line 221
at Docker.Backend.ContainerEngine.Windows.Start(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Windows.cs:line 93
at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.b__0(Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 47
at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 145

Tried both stable and edge version, settings are default.
[log.txt](https://github.com/docker/for-win/files/1646985/log.txt)

**Comment by jasonbivins** [comment]
Hi @dreamflame51 
There are some network errors in the logs coming from HNS - but they are unspecified errors.  Have you tried rebooting?  
```[16:44:04.958][WindowsDaemon  ][Error  ] fatal: Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : Unspecified error
[16:44:04.958][WindowsDaemon  ][Error  ] fatal: Error starting daemon: Error initializing network controller: Error creating default network: HNS failed with error : Unspecified error```

**Comment by dreamflame51** [comment]
Hi jasonbivins,

Sure, I did.. Rebooted, reinstalled (edge/stable)

**Comment by acerock** [comment]
Had the same problem and went through the same pain with install/reinstall, remove/add hyper-v, etc. I found the problem related to nat networking and the vSwitch and docker for some reason can't make the Network Host Service to apply the required settings.
For me it helped to do:
	Stop-Service hns -Force
	del C:\ProgramData\Microsoft\Windows\HNS\HNS.data
	Start-Service hns -Force
	<reboot>

Prior to this (without any immediate effect) I also did:
	Get-NetNat | Remove-NetNat
Don't know if this actually helped, but it removed one item from my computer.

I'm now running 18.01.0-ce (Wed Jan 10 2018) with experimental features on windows 10 fall creators update (1709) using windows containers.

Helpful links for me was:
https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/setup-nat-network#troubleshooting
https://github.com/MicrosoftDocs/Virtualization-Documentation/tree/live/windows-server-container-tools/CleanupContainerHostNetworking
